rasa/telemetry.py

```diff
@@ -631,7 +634,19 @@ def initialize_error_reporting() -> None:
         ],
         send_default_pii=False,  # activate PII filter
         server_name=telemetry_id or "UNKNOWN",
-        ignore_errors=[KeyboardInterrupt, RasaException, NotImplementedError],
+        ignore_errors=[
+            # std lib errors
+            KeyboardInterrupt,
+            MemoryError,
+            NotImplementedError,
+            asyncio.CancelledError,
+            # rasa errors
+            RasaException,
+            # 3rd party errors
+            botocore.exceptions.NoCredentialsError,
```

Comment:
I think we should ignore more connection errors too (see [all the errors here](https://github.com/boto/botocore/blob/develop/botocore/exceptions.py))